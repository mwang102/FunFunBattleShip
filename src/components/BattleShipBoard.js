import React from 'react'
import styled from "styled-components";
import Row from './Row'
import PropTypes from 'prop-types'


export default class BattleShipBoard extends React.Component{
	static propTypes = {
		handlePick:PropTypes.func.isRequired,
	}

	render(){
		const table = this.props.board.map((row, index)=>{
			return <Row handlePick={this.props.handlePick} rowNumber={index} key={index} data={row}/>
		})

		return (
			<BoardContainer> {table} </BoardContainer>
		)
	}

}


const BoardContainer = styled.section`
	display:flex;
	flex:4;
	flex-direction:column;
	height:auto;
	margin:1rem;
	max-width:1000px;
`;