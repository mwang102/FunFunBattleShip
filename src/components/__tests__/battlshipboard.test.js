import React from 'react';
import { shallow } from 'enzyme'
import BattleShipBoard from '../BattleShipBoard'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('BattleShipBoard', ()=>{
	it('check for row length', ()=>{
		const mockTable = [
        [[5,false],[0,false],[0,false],[0,false]],
        [[5,false],[0,false],[0,false],[0,false]],
        [[5,false],[2,false],[2,false],[0,false]],
        [[0,false],[0,false],[0,false],[0,false]]
      ]

		const mockCallback = jest.fn()

		const wrapper = shallow(<BattleShipBoard handlePick={mockCallback} board={mockTable}/>)

		const rows = wrapper.find('Row')
		expect(rows.length).toEqual(4)
	})

	it('matches the snapshot', ()=>{
		const mockTable = [
        [[5,false],[0,false],[0,false],[0,false]],
        [[5,false],[0,false],[0,false],[0,false]],
        [[5,false],[2,false],[2,false],[0,false]],
        [[0,false],[0,false],[0,false],[0,false]]
      ]

		const mockCallback = jest.fn()
		const tree = renderer.create(<BattleShipBoard handlePick={mockCallback} 
													  board={mockTable}/>).toJSON()
		
		expect(tree).toMatchSnapshot()
	})
})