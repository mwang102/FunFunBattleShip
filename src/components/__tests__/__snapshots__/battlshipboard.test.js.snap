// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BattleShipBoard matches the snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px 5px;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 0px 1.0875rem 1.45rem;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.5);
  font-size: 30px;
  background-repeat: no-repeat;
  background-position: center;
  background: BlanchedAlmond;
  cursor: pointer;
}

.c2:hover {
  background: gold;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 5px;
  background: ivory;
  height: auto;
  -webkit-animation: cJUYLg;
  animation: cJUYLg;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: .0s;
  animation-delay: .0s;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 5px;
  background: ivory;
  height: auto;
  -webkit-animation: cJUYLg;
  animation: cJUYLg;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: .1s;
  animation-delay: .1s;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 5px;
  background: ivory;
  height: auto;
  -webkit-animation: cJUYLg;
  animation: cJUYLg;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: .2s;
  animation-delay: .2s;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 5px;
  background: ivory;
  height: auto;
  -webkit-animation: cJUYLg;
  animation: cJUYLg;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: ease;
  animation-timing-function: ease;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation-delay: .3s;
  animation-delay: .3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 4;
  -ms-flex: 4;
  flex: 4;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  margin: 1rem;
  max-width: 1000px;
}

<section
  className="c0"
>
   
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
  </div>
  <div
    className="c3"
  >
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
  </div>
  <div
    className="c4"
  >
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
  </div>
  <div
    className="c5"
  >
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
    <div
      className="c2"
      onClick={[Function]}
    />
  </div>
   
</section>
`;
